backend/server.js

```diff
@@ -134,39 +180,84 @@ function addBackendRoutes (app, config, store) {
      grants. */
   app.post('/upload', function (req, res) {
     config.getUserConfig(req, function (err, userConfig) {
-      selectTarget (userConfig, req.body, function (err, target) {
-        if (err) return res.json({error: err.toString()});
+      selectTarget(userConfig, req.body, function (err, target) {
+        if (err) return res.json({ error: err.toString() });
         const s3client = upload.makeS3UploadClient(target);
-        const {s3Bucket, uploadPath: uploadDir} = target;
+        const { s3Bucket, uploadPath: uploadDir } = target;
         const id = Date.now().toString();
         const uploadPath = `${uploadDir}/${id}`;
         upload.getJsonUploadForm(s3client, s3Bucket, uploadPath, function (err, events) {
-          if (err) return res.json({error: err.toString()});
+          if (err) return res.json({ error: err.toString() });
           upload.getMp3UploadForm(s3client, s3Bucket, uploadPath, function (err, audio) {
-            if (err) return res.json({error: err.toString()});
+            if (err) return res.json({ error: err.toString() });
             const baseUrl = `https://${s3Bucket}.s3.amazonaws.com/${uploadPath}`;
             const player_url = `${config.playerUrl}?base=${encodeURIComponent(baseUrl)}`;
-            res.json({player_url, events, audio});
+            if (!config.enableOauth && config.database) 
+              mysqlUtils.storeRecord(req.session.userId, baseUrl, id, config.database);
+            res.json({ player_url, events, audio });
           });
         });
       });
     });
   });
 
+  app.post('/delete', checkLogin, function (req, res) {
+
+    let recordId = req.body.recordId;
+    if (req.session.context.isAdmin) {
+      deleteRecord(true);
+    } else {
+      mysqlUtils.userHavePrivileges(
+        recordId,
+        req.session.userId,
+        config.database,
+        deleteRecord
```

Comment:
What if we use an anonymous function here what will get the boolean if user has the permission or not. 
Then we can delete the record or redirect to an error page accordingly? This way `deleteRecord`  will only be deleting records.
If this seems suitable to you, we can use this for other such utility functions too

Reply:
yes that looks good, applied.